include/aikido/planner/ConfigurationToConfigurations.hpp

```diff
@@ -47,14 +47,14 @@ class ConfigurationToConfigurations : public Problem
   std::size_t getNumGoalStates() const;
 
   /// Returns goal states.
-  const GoalStates& getGoalStates() const;
+  const GoalStates getGoalStates() const;
 
 protected:
   /// Start state.
-  const statespace::StateSpace::State* mStartState;
+  statespace::StateSpace::ScopedState mStartState;
```

Comment:
Why remove the `const`? Looks like we are updating the member in the initializer list, so should be fine? 

That said, looks like the other `problem`s have it non-const, we can also address all the const-ness in another PR. Could you make an issue?

Reply:
Yeah, for some reason there are issues with the `ScopedState` class that make compilation fail if we do that (I think that's why all of the other classes have it as non-const as well). Fixing that looks like it's a big task (probably outside the scope of this PR).

Once we merge I'm happy to make an issue for us to go back and clean this up later! :smile: